src/frontend/src/optimizer/plan_node/stream_local_simple_agg.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use itertools::Itertools;
+use risingwave_common::catalog::{DatabaseId, SchemaId};
+use risingwave_pb::stream_plan::stream_node::NodeBody as ProstStreamNode;
+
+use super::logical_agg::PlanAggCall;
+use super::{LogicalAgg, PlanBase, PlanRef, PlanTreeNodeUnary, ToStreamProst};
+use crate::optimizer::property::RequiredDist;
+
+#[derive(Debug, Clone)]
+pub struct StreamLocalSimpleAgg {
+    pub base: PlanBase,
+    logical: LogicalAgg,
+}
+
+impl StreamLocalSimpleAgg {
+    pub fn new(logical: LogicalAgg) -> Self {
+        let ctx = logical.base.ctx.clone();
+        let pk_indices = logical.base.pk_indices.to_vec();
+        let input = logical.input();
+        let input_dist = input.distribution();
+        debug_assert!(input_dist.satisfies(&RequiredDist::AnyShard));
+
+        // Single-value local simple agg only produces inserts
+        // it should be append-only, even if input is non-append.
+        // N.B. This assumes input is valid - non-append-only min/max is rejected.
+        let append_only = true;
```

Comment:
The semantics of local simple agg is somehow strange (it cannot be materialized). So I'd prefer to set append_only false?

Reply:
Hmm I used it for this purpose: https://github.com/singularity-data/risingwave/pull/3320#discussion_r906204180.

TLDR: global agg uses single value state if input is `append_only`. Otherwise it may use `extreme` agg state e.g. for `min/max`. That means we can't support append_only min/max.

Perhaps we can change `append_only` to `append_only_output`? I'll try to think of some alternatives.

Reply:
We can defer `append_only` min/max for now I guess. Since stateful approach will deal with it.